lib/supa/command.rb

```
@@ -15,42 +15,59 @@ def represent
 
     private
 
-    def value
-      return instance_exec(&value_accessor) if value_accessor.respond_to?(:call)
+    attr_reader :representer, :context, :tree, :name, :options, :block
+
+    def apply_modifier(value)
+      with_modifier? ? representer.send(modifier, value) : value
+    end
 
-      extracted_value = derived_value_from_object(@object) || derived_value_from_object(@representer)
-      extracted_value ||= literal_value
+    def modifier
+      options[:modifier]
+    end
 
-      raise_no_method_error(value_accessor) if extracted_value.nil?
-      extracted_value
+    def with_modifier?
+      !options[:modifier].nil?
     end
 
-    def value_accessor
-      @value_accessor ||= @options.fetch(:getter, @name)
+    def static_value
+      getter
     end
 
-    def derived_value_from_object(object)
-      if value_accessor.respond_to?(:to_sym) && object.respond_to?(value_accessor)
-        object.send(value_accessor.to_sym)
-      elsif object.is_a?(Hash)
-        object.dig(value_accessor)
+    def dynamic_value
+      if exec_on_object?
+        value_from_object
+      else
+        value_from_representer
       end
     end
 
-    def literal_value
-      return if value_accessor.respond_to?(:call)
-      value_accessor unless value_accessor.is_a?(Symbol) || value_accessor.is_a?(Enumerable)
+    def exec_on_object?
+      options[:exec_context] != :representer
+    end
+
+    def value_from_object
+      return context[getter] if context.is_a?(Hash)
+      return context.send(getter) if context.respond_to?(getter)
```

needs more input validation: it's easy to fail `respond_to?` with an error if `getter` is not what expected

---

I would like to raise an Exception rather than not returning anything if we don't have a Hash and the method is not defined on the object